src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -402,6 +409,14 @@ static function ( $format ) {
 		// Force the post_type argument, since it's not a user input variable.
 		$args['post_type'] = $this->post_type;
 
+		if ( Hierarchical_Sort::is_eligible( $request ) ) {
+			$result       = Hierarchical_Sort::run( $args );
+			$this->levels = $result['levels'];
```

Comment:
@oandregal: Correct me if I'm wrong, but this is [the same problem](https://github.com/WordPress/gutenberg/pull/66479#discussion_r1883888204) that we had with static variables in the original Gutenberg PR. The concept of posts controller doesn't map 1:1 to the concept of a query, and so attaching the state of a query (`levels`) to a controller instance that can outlive it (`$this`) seems dangerous.